### PR TITLE
EMS106 - HR can show particular employee attendance

### PR DESCRIPTION
**Pull request by rorpyll**

### CODE REVIEW

File: app/controllers/attendences_controller.rb [hunk 1]
```diff
@@ -4,27 +4,33 @@
...
+                  # @attendences = Attendence.where(params[:employee_id])
+                  else
+                    Attendence.find_by(employee_id: current_employee.id)
+                  end
+    @attendences = Attendence.where(employee_id: @attendence.employee_id)
   end
 
   def edit
-    @attendence = Attendence.find(params[:id])
+    if current_employee.role.name == 'HR'
```
Comment (rorshaan) [inline app/controllers/attendences_controller.rb:21]: Add a method to check the role and replace it everywhere where you are checking for HR and other roles

**Review by rorshaan — CHANGES_REQUESTED** [review]
Test cases are missing.

File: app/controllers/attendences_controller.rb [hunk 2]
```diff
@@ -2,29 +2,36 @@
 
 # Attendence controller
 class AttendencesController < InheritedResources::Base
+  include ApplicationHelper
```
Comment (rorshaan) [inline app/controllers/attendences_controller.rb:5]: Why application helper is added to to controller.
Create a saperate module and include it

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -14,4 +14,9 @@ def salary_total
     #   return s.salary / 30 * array.uniq.count
     # end
   end
+
+
+  def employee_role
```
Comment (rorshaan) [inline app/helpers/application_helper.rb:19]: Define it in model and use proper naming convention while defining method names

File: app/controllers/tickets_controller.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class TicketsController < InheritedResources::Base
+  include ApplicationHelper
   def index 
-    if current_user.role.name == 'HR'
+    if employee_role
```
Comment (rorshaan) [inline app/controllers/tickets_controller.rb:4]: what is the meaning of employee role?

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -26,6 +26,7 @@ class ApplicationController < ActionController::Base
   #    	end
   #    end
 
+
```
Comment (salvimohan) [inline app/controllers/application_controller.rb:29]: why extra line

File: app/views/employees/show.html.erb [hunk 6]
```diff
@@ -1,7 +1,7 @@
  </aside><!-- End Sidebar-->
 
   <main id="main" class="main">
-<% if current_employee == @employee || current_employee.role.name == 'HR' %>
+<% if current_employee == @employee || is_hr? %>
```
Comment (salvimohan) [inline app/views/employees/show.html.erb:4]: it should be current_employee.is_hr?

File: app/controllers/employees_controller.rb [hunk 7]
```diff
@@ -14,34 +14,33 @@ def index
...
+                  Employee.find(params[:id])
+                else
+                  current_user
+                end
   end
 
-  def edit 
-    @designations = Designation.all 
+  def edit
+    @designations = Designation.all
```
Comment (salvimohan) [inline app/controllers/employees_controller.rb:25]: we can create a before action and move this three line there

File: app/controllers/leafs_controller.rb [hunk 8]
```diff
@@ -1,39 +1,42 @@
...
   def index
-    if current_employee.is_hr?
-      @leafs = Leaf.order('created_at DESC')
-    else
-      @leafs = Leaf.where(employee_id: current_employee.id).order('created_at DESC')
-    end
+    @leafs = if current_employee.is_hr?
+               Leaf.order('created_at DESC')
+             else
+               Leaf.where(employee_id: current_employee.id).order('created_at DESC')
```
Comment (salvimohan) [inline app/controllers/leafs_controller.rb:10]: We can user current_employee.leaves here

**Pull request closed without merging**